docs/requirements.txt

```diff
@@ -1,6 +1,6 @@
 sphinx==5.1.1
 sphinx-rtd-theme==1.0.0
 # dependencies from setup.py
-py-algorand-sdk>=1.9.0,<2.0.0
+py-algorand-sdk>=1.16.1,<2.0.0
```

Comment:
I briefly checked the changelog of py-sdk, and 1.16.1 should contain the most recent changes to the part PyTeal need from SDK, so any version >= 1.16.1 should suffice. 1.20.0 doesn't seem much different from 1.16.1 on PyTeal side (for now), I think.

Reply:
What new features/changes in the SDK necessitate increasing the version?

Reply:
Honestly, this is forward looking to a requirement in a future PR (for source mapping). And philosophically, with PyTeal allowing boxes functionality, we _ought_ to depend on a PySDK release that has the same. However, we can probably get away with not changing this dependency at all and still have CI passing. If there is a strong desire/argument for maintaining the 1.9.0 lower bound, I can see how far I can push graviton's dependency and still pass Ci.

Also @michaeldiamant always has valuable suggestions regarding versioning. Any thoughts?

Reply:
Chiming in per the above request with my 2c:
* Since the highlighted section of `docs/requirements.txt` intends to duplicate `setup.py`, I'd continue following the convention.  So, unless `setup.py` is changed, I'd make no change here.
* For `setup.py`, I've based my understanding on https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#id5.  My understanding is to leave `setup.py` as loose as possible.  Since someone may use PyTeal _without_ using Boxes, I think the version can be left unchanged.

Reply:
Ok, I'll revert the update to pysdk and follow through on any implications.

I assume that the other minor bumps of dependencies are less controversial. But folks should speak up if they disagree.